### PR TITLE
Dotenv Customization

### PR DESCRIPTION
**Pull request by colynb**
The branch includes changes that will allow developers to completely customize the dotenv library.

The parsing of env vars now happens inside a new file called `dotenv.config.js`. If you want to override this behavior, for instance, you have some additional plugins and other parsing requirements, just create your own dotenv.config.js in the root of your project and the serverless-dotenv-plugin will use that instead.

### CODE REVIEW

**Comment by kf6kjg** [comment]
The idea seems sound, however I would note that my codebases are 100% TypeScript - I've even taken pains to make sure my webpack config files are in TypeScript.  I'd suggest that if this library goes down the road specified by this PR that it provide a way to handle that case.

Here are the webpack docs on the subject as a reference for one method of doing this: https://webpack.js.org/configuration/configuration-languages/

File: index.js [hunk 1]
```diff
@@ -35,18 +37,33 @@ class ServerlessPlugin {
...
+    let pluginConfigPath = path.resolve(basePath, pluginConfigName)
+
+    return fs.existsSync(pluginConfigPath)
+      ? pluginConfigPath
+      : defaultPluginConfigPath
   }
 
   loadEnv(env) {
     var envFileName = this.resolveEnvFileName(env)
+    console.log(envFileName)
```
Comment (bryandowning) [inline index.js:62]: stray console.log here

**Comment by neverendingqs** [comment]
> The idea seems sound, however I would note that my codebases are 100% TypeScript - I've even taken pains to make sure my webpack config files are in TypeScript. I'd suggest that if this library goes down the road specified by this PR that it provide a way to handle that case.
> 
> Here are the webpack docs on the subject as a reference for one method of doing this: https://webpack.js.org/configuration/configuration-languages/

That seems like a big lift given this project isn't written in TypeScript. Even if it is, if I'm understanding correctly, this plugin will have to compile the TypeScript into JavaScript on-the-fly to use the config.

Alternatively, could we make the path to `dotenv.config.js` customizable? That way, you can point to a compiled version of your config.

**Comment by kf6kjg** [comment]
Most of my code isn't compiled to disk: instead the needed subsystems are registered with the tools on the command line. 

That this project isn't in typescript isn't a problem to me, no uplift needed. To me this is an architectural issue. One that has been solved in other projects: see the -r flag used by node and many others that allows typescript, and many other things, to be handled. And it's probably not the only solution, just one I'm familiar with.

**Comment by neverendingqs** [comment]
> To me this is an architectural issue. One that has been solved in other projects

I would say that what you are asking is a feature, not an issue.

Not all JavaScript libraries need to support TypeScript IMO, but I'm guessing this is where we disagree :).

**Comment by kf6kjg** [comment]
> a feature, not an issue.
Touché.

Need to? No. Would I prefer? Yes. I too maintain a couple of open source
projects and nothing turns me off faster than being told I have to do
something. My point was to make my opinion known do that it can be
considered: one can’t make an informed decision of the information is
lacking. :)

So yes, back to the point, I would like if there was provided a way to read
the config file in typescript if a feature like was proposed in this PR is
considered by whomever takes up this banner.

**Pull request closed without merging**